crypto/crypto/src/multi_ed25519.rs

```diff
@@ -0,0 +1,547 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides an API for the accountable threshold multi-sig PureEdDSA signature scheme
+//! over the ed25519 twisted Edwards curve as defined in [RFC8032](https://tools.ietf.org/html/rfc8032).
+//!
+//! Signature verification also checks and rejects non-canonical signatures.
```

Comment:
could you document your framework / interface up here a bit more clearly or if it is easier in the description fo the primary structures

Reply:
rust doc tests are a great way to do this.

Reply:
Planning to do that, I was hoping to split tests in multiple PRs to keep this as small as possible. Is this ok?